### ISSUE TITLE
v8 build is broken

### ISSUE THREAD

**Issue by lazka** [msys2/MINGW-packages#12568]
I suspect because the CI image now comes with vs2022 and previously it found vs2019 and carried on. msvc checks should probably be patched out somehow, it's not obvious to me how though, so leaving this broken for now.

```
[280/280] LINK gn_unittests.exe
Traceback (most recent call last):
  File "C:\M\mingw-w64-v8\src\v8-9.1.269.39\build\vs_toolchain.py", line 573, in <module>
    sys.exit(main())
  File "C:\M\mingw-w64-v8\src\v8-9.1.269.39\build\vs_toolchain.py", line 569, in main
    return commands[sys.argv[1]](*sys.argv[2:])
  File "C:\M\mingw-w64-v8\src\v8-9.1.269.39\build\vs_toolchain.py", line 546, in GetToolchainDir
    runtime_dll_dirs = SetEnvironmentAndGetRuntimeDllDirs()
  File "C:\M\mingw-w64-v8\src\v8-9.1.269.39\build\vs_toolchain.py", line 106, in SetEnvironmentAndGetRuntimeDllDirs
    os.environ['GYP_MSVS_OVERRIDE_PATH'] = DetectVisualStudioPath()
  File "C:\M\mingw-w64-v8\src\v8-9.1.269.39\build\vs_toolchain.py", line 197, in DetectVisualStudioPath
    version_as_year = GetVisualStudioVersion()
  File "C:\M\mingw-w64-v8\src\v8-9.1.269.39\build\vs_toolchain.py", line 186, in GetVisualStudioVersion
    raise Exception('No supported Visual Studio can be found.'
Exception: No supported Visual Studio can be found. Supported versions are: 16.0 (2019), 15.0 (2017).
Package icu-i18n was not found in the pkg-config search path.
Perhaps you should add the directory containing `icu-i18n.pc'
to the PKG_CONFIG_PATH environment variable
Package 'icu-i18n', required by 'virtual:world', not found
Could not run pkg-config.
Package zlib was not found in the pkg-config search path.
Perhaps you should add the directory containing `zlib.pc'
to the PKG_CONFIG_PATH environment variable
Package 'zlib', required by 'virtual:world', not found
Could not run pkg-config.
ERROR at //build/config/linux/pkg_config.gni:117:17: Script returned non-zero exit code.
    pkgresult = exec_script(pkg_config_script, args, "value")
                ^----------
Current dir: C:/M/mingw-w64-v8/src/v8-9.1.269.39/out/Release-i686/
Command: D:/a/msys64/mingw32/bin/python.exe C:/M/mingw-w64-v8/src/v8-9.1.269.39/build/config/linux/pkg-config.py icu-i18n
Returned 1.
See //third_party/icu/BUILD.gn:12:1: whence it was called.
pkg_config("system_icui18n") {
^-----------------------------
See //BUILD.gn:1624:21: which caused the file to be included.
    public_deps = [ "//third_party/icu" ]
                    ^------------------
==> ERROR: A failure occurred in build().
```

**Comment by lazka** [comment]
@Kwizatz hey, are you still interested in the v8 package?

**Comment by naveen521kk** [comment]
I'll have a look at this a bit later, mostly it looks similar to skia package which I'm familiar with.

**Comment by Kwizatz** [comment]
> @Kwizatz hey, are you still interested in the v8 package?

I am, but is hard to keep up. As it is I have to make it work on VCPKG first then re-add all the MinGW patches, update repo SHAs and so on. merging the changes upstream is likely not happening, because there are changes required on the main Chromium branch (the build directory), which is a whole other bag of worms.

So I will probably just drop VCPKG given that they don't want the patches there to have any MinGW changes, and the latest upgrade is not passing CI because of the same thing, the required SDK not being installed on the CI VMs.

Anyway, if you don't mind, I'll make my fork a proper fork and either build the PKGs directly from that OR extract the patches for the PKGBUILD.

**Comment by Kwizatz** [comment]
... Actually:

[280/280] LINK gn_unittests.exe

Seems to be a problem with the GN build, you can probably remove that part since the MSYS/MINGW package for GN must already be patched to work fine.

**Comment by lazka** [comment]
@Kwizatz no worries. Whatever is easier for you.

Fixed by #12587 for now
